### ISSUE TITLE
Once again a CPU (Multicore) Question...

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#810]
Another vert strange observation here regarding mutlithreading...

The Core setting in Radiums options doesn't make any difference. Like at all.
Only when going to 1 core it goes up by 10%. I am sure something weird is going on with the multithreading.

Multicore Setting CPU 2 (Two):      45/47/57% 
Multicore Setting CPU 6 (Six):        42/48/58
Multicore Setting CPU 12 (Twelve): 45/50/59 
One Core Only: 73/78/ +- 110 (Crackle starting) 
(When using one core only: in the OSX System we are still running on multiple cores, but all just higher.
Doesn't make any sense to me. Can you explain?)

I crosschecked these numbers against the OSX system itself, they are all 
basically more or less correct.

Huh? :)

Best,
tL.

**Comment by kmatheussen** [comment]
The multicore scheduler in Radium is very light weight. It doesn't
buzy-wait, like probably all the big DAWs does. This means that
the program uses less CPU overall, but also that multicore doesn't
work so well. I should probably change that (when I experienced
with buzy-looping, I got far better performance). I guess this is the
most likely explanation for these numbers.

**Comment by Lykdraft** [comment]
Alright, I see. Thanks.

**Comment by kmatheussen** [comment]
I think the ticket should be open. It shouldn't be that much work to improve the scheduler.

**Comment by Lykdraft** [comment]
Alright, sounds good. Looking forward to it.

**Comment by kmatheussen** [comment]
Can you check if 4.4.3 improves the situation?

**Comment by Lykdraft** [comment]
Did check. Unfortunatley no difference between a setting of 2 and a setting of 12.
(In a full song with 60 modules. So there should be something usually I guess..)

Just a setting of 1 core only doubles the CPU consumption. All other settings make no difference.
Best,
tL.

**Comment by kmatheussen** [comment]
I've tested this quite a bit, both when I made the original
multi core scheduler, and also yesterday, and it does scale
beyond 2 cores.

My guess is that your song doesn't have an audio graph with
more than 2 paths which the program is able to run
in parallel very well. (It is possible to work around this,
but not without introducing extra delays in the graph.)

**Comment by kmatheussen** [comment]
Hmm, so far I haven't been able to scale it beyond two cpus
in release build. I just tested with debug build on tuesday.
I guess it scales for instruments that uses a lot of CPU though.

**Comment by Lykdraft** [comment]
Again, I've tested this on a full produced and already released song.
There is tons of stuff going on. When I open the mixer window I see
tons of channel strips, so there should absolut be enough audio-graphs... ;)

**Comment by kmatheussen** [comment]
Thank you. I guess the scheduler is not smart enough.
I'm doing some improvements now.

**Comment by kmatheussen** [comment]
There's been some improvements to the scheduler. (well, actually, it's competely rewritten again, but I couldn't measure gain in performance. :-) )

But I've tested a bit more, and the scheduler doesn't stop at 2 cores. Your song probably just has one path that uses equally much, or more, CPU as all the other paths combined. When that happens, it doesn't matter how many CPU the computer has.

Closing issue now.
